csrc/prepare_inputs/advance_step.cu

```diff
@@ -0,0 +1,126 @@
+/*
+ * TODO: Add doc
+ */
```

Comment:
can we clean up the PR? looks like three unresolved TODOs

Reply:
given this kernel is currently specialized to spec decode, consider linking back to the PR

Reply:
Added